middleware/validate.js

```diff
@@ -0,0 +1,168 @@
+module.exports = {
+  validator: function (req, res, next) {
+
+    /************************
+     *  ARTICLES
+    ************************/
+    if (req.originalUrl.slice(0, 9) === '/articles') {
+      const title = req.params.title;
+      const articleDB = require('../db/articles.js');
+      const database = articleDB.getArticles().articles;
```

Comment:
You shouldn't try to use the array inside articleDb like this. You should stick with the methods expose through the module.exports.